### PR TITLE
slate build: fix Py3 byte+str concat errors in testsweeper subrepo

### PR DESCRIPTION
**Pull request by eugeneswalker**
Package `slate` relies on sub-repo `testsweeper`, which uses a Python configure script that has byte+str concatenation errors when run with Python 3. This PR uses `filter_file` on the `testsweeper` configure script to mend the offending lines (4 lines in total). Obviously it would be preferable to have an upstream fix in the `testsweeper` repo but for now this allows `slate` to build.

Temporarily addresses https://github.com/spack/spack/issues/15216

@G-Ragghianti

### CODE REVIEW

File: var/spack/repos/builtin/packages/slate/package.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ def install(self, spec, prefix):
 
         compiler = 'mpicxx' if spec.variants['mpi'].value else ''
 
+        configfile = '{}/testsweeper/config/config.py'.format(self.stage.source_path)
+        filter_file('log\.write\( stdout \)','log.write( str(stdout) )', configfile, backup=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/slate/package.py]:
```suggestion
        filter_file('log.write( stdout )', 'log.write( str(stdout) )', configfile, backup=False, string=True)
```
Same below, plus line wrapping to make flake8 happy

File: var/spack/repos/builtin/packages/slate/package.py [hunk 2]
```diff
@@ -37,13 +37,31 @@ def setup_build_environment(self, env):
...
+                    configfile, backup=False, string=True)
+
     def install(self, spec, prefix):
         f_cuda = "1" if spec.variants['cuda'].value else "0"
         f_mpi = "1" if spec.variants['mpi'].value else "0"
         f_openmp = "1" if spec.variants['openmp'].value else "0"
 
         compiler = 'mpicxx' if spec.variants['mpi'].value else ''
 
+        self.patch()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/slate/package.py]: No need to call this, it’s automatically called before all other stages.

**Comment by mgates3** [comment]
We're fixing this in TestSweeper, as well as writing a new Spack package for SLATE itself. Fixes expected within next couple weeks.

**Comment by eugeneswalker** [comment]
> We're fixing this in TestSweeper, as well as writing a new Spack package for SLATE itself. Fixes expected within next couple weeks.

That is great! I wonder if it is best to wait for the upstream fix in that case? Any thoughts? If we don't wait, we can always commit this in the meantime. @mgates3

**Comment by mgates3** [comment]
SLATE has been updated with the newer TestSweeper. It should be working with both python2 and python3.

**Comment by adamjstewart** [comment]
@eugeneswalker since this is fixed in upstream and since this package only contains a single develop version, I take it this PR is no longer needed?

**Comment by eugeneswalker** [comment]
> @eugeneswalker since this is fixed in upstream and since this package only contains a single develop version, I take it this PR is no longer needed?

Yes. I agree. No longer needed. Thanks!

**Pull request closed without merging**